### PR TITLE
feat: add new suggestion

### PR DESCRIPTION
**Pull request by ManuViola77**
## Links:

[Agregar una pagina de sugerencias/feedback para que la gente pueda proponer mejoras y votarlas](https://github.com/users/ManuViola77/projects/1/views/1#:~:text=Agregar%20una%20pagina%20de%20sugerencias/feedback%20para%20que%20la%20gente%20pueda%20proponer%20mejoras%20y%20votarlas)

## What & Why:

This PR adds the possibility to add new suggestions. In next PRs I will add the delete/edit suggestion functionality.

## Risks, Mitigation & Rollback:

- [ ] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/hooks/useSuggestions.js [hunk 1]
```diff
@@ -1,94 +1,169 @@
...
+  const { title, description } = newSuggestion;
 
-  const {
-    user: { uid: myId },
-  } = useAuth();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const handleOpenModal = () => setIsModalOpen(true);
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+    setErrors(EMPTY_SUGGESTION);
```
Comment (ManuViola77) [inline src/hooks/useSuggestions.js:64]: this should be `EMPTY_ERRORS`, I fix that in the next PR

File: src/hooks/useSuggestions.js [hunk 2]
```diff
@@ -1,94 +1,169 @@
...
-    }));
+  useEffect(() => {
+    getSuggestions();
+  }, [getSuggestions]);
+
+  const onChangeFilter = (key, newValue) => handleOnChangeState(setFilters, key, newValue);
+
+  const onChangeNewSuggestion = (key, newValue) => {
+    handleOnChangeState(setNewSuggestion, key, newValue);
+    setErrors(EMPTY_SUGGESTION);
```
Comment (ManuViola77) [inline src/hooks/useSuggestions.js:125]: same here

File: src/hooks/useSuggestions.js [hunk 3]
```diff
@@ -1,94 +1,169 @@
...
+  const [isLoading, setIsLoading] = useState(false);
   const [suggestions, setSuggestions] = useState([]);
+  const [errors, setErrors] = useState(EMPTY_ERRORS);
+  const [newSuggestion, setNewSuggestion] = useState(EMPTY_SUGGESTION);
   const [filters, setFilters] = useState({ statusFilter: DEFAULT_STATUS });
 
   const {
     statusFilter: { value: statusFilterValue },
   } = filters;
+  const { title, description } = newSuggestion;
```
Comment (jeisux92) [inline src/hooks/useSuggestions.js]: usememo

File: src/pages/Suggestions/styles.css [hunk 4]
```diff
@@ -13,3 +21,45 @@
...
+}
+
+.field-label {
+  font-weight: 800;
+  padding-bottom: 10px;
+}
+
+.add-suggestion-button {
+  margin: 10px auto;
+  max-width: 12rem;
```
Comment (jeisux92) [inline src/pages/Suggestions/styles.css]: do not mix rem with px

File: src/hooks/useSuggestions.js [hunk 5]
```diff
@@ -1,94 +1,169 @@
...
+    if (!description) {
+      onChangeErrors('description', t('errors.emptyDescription'));
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      await addDoc(suggestionsRef, newSuggestion);
+      getSuggestions();
```
Comment (jeisux92) [inline src/hooks/useSuggestions.js]: add await

File: src/pages/Suggestions/styles.css (hunk 4 above)
Comment (ManuViola77) [reply src/pages/Suggestions/styles.css]: thanks I think I dont need this, I can delete it 😄

**Review by jeisux92 — APPROVED** [review]
lgtm

**Pull request merged**